hibikase/PlaybackWidget.cpp

```diff
@@ -110,30 +146,49 @@ void PlaybackWidget::OnStateChanged(QAudio::State state)
         m_play_button->setText(QStringLiteral("Stop"));
 
     if (state == QAudio::State::StoppedState)
-        UpdateTime(std::chrono::milliseconds(-1));
+        UpdateTime(std::chrono::microseconds(-1), std::chrono::microseconds(-1));
 }
 
-void PlaybackWidget::UpdateTime(std::chrono::milliseconds ms)
+void PlaybackWidget::UpdateTime(std::chrono::microseconds current, std::chrono::microseconds length)
 {
     QString text;
     // When we request to delete the worker, it doesn't happen immediately;
     // if it triggers time updates in that period we want to ignore them.
-    if (m_worker && ms.count() >= 0)
+    if (m_worker && current.count() >= 0)
     {
-        long long time = ms.count();
-        text = QStringLiteral("%1:%2:%3").arg(time / 60000,     2, 10, QChar('0'))
-                                         .arg(time / 1000 % 60, 2, 10, QChar('0'))
-                                         .arg(time / 10 % 100,  2, 10, QChar('0'));
+        long long cur = current.count() / 1000;
+        long long len = length.count() / 1000;
+        text = QStringLiteral("%1:%2:%3 / %4:%5:%6").arg(cur / 60000,     2, 10, QChar('0'))
+                                                    .arg(cur / 1000 % 60, 2, 10, QChar('0'))
+                                                    .arg(cur / 10 % 100,  2, 10, QChar('0'))
+                                                    .arg(len / 60000,     2, 10, QChar('0'))
+                                                    .arg(len / 1000 % 60, 2, 10, QChar('0'))
+                                                    .arg(len / 10 % 100,  2, 10, QChar('0'));
```

Comment:
You could split out the xx:xx:xx formatting code to a function and call it twice.